# The reverse complement of Chr7 43,600,000-43,750,000  inserted into Chr7 43,600,000

### Analysis of the genome in which: 

The reverse complement of **chr7 43,600,000-43,750,000** is insert in the the region **chr7 43,600,000-43,610,000**

### Primary region spanning: chr7 43,600,000-43,750,000 

For this analysis the split reads at chr7 43,590,000 and 43,600,000 were selected and analysed using the ___Variant determination___ > ___Use soft clip data___ > ___Complex rearrangement___ menu option.<hr />

![image](images/insert_chr7_60_43,600,000-43,750,000_RC_target_chr7_60_43,600,000-43,610,000_ONT_no_2nd_1.jpg)

Figure 1

<hr />

![image](images/insert_chr7_60_43,600,000-43,750,000_RC_target_chr7_60_43,600,000-43,610,000_ONT_no_2nd_1_all.jpg)

Figure 2

<hr />

![image](images/insert_chr7_60_43,600,000-43,750,000_RC_target_chr7_60_43,600,000-43,610,000_ONT_no_2nd_1_result.jpg)

Figure 3

<hr />

### Primary region spanning: chr7 43,600,000-43,750,000 

For this analysis the split reads at chr7 43,600,000 whose secondary alignments mapped to chr7 43,750,000 (see red line in Figure 4 and 5) and 43,600,000 whose secondary reads mapped to chr7 43,6100,00 (see green line in Figure 4 and 5) were selected and analysed using the ___Variant determination___ > ___Use soft clip data___ > ___Complex rearrangement___ menu option.<hr />

![image](images/insert_chr7_60_43,600,000-43,750,000_RC_target_chr7_60_43,600,000-43,610,000_ONT_no_2nd_2.jpg)

Figure 4

<hr />

![image](images/insert_chr7_60_43,600,000-43,750,000_RC_target_chr7_60_43,600,000-43,610,000_ONT_no_2nd_2_all.jpg)

Figure 5

<hr />

![image](images/insert_chr7_60_43,600,000-43,750,000_RC_target_chr7_60_43,600,000-43,610,000_ONT_no_2nd_2_result.jpg)

Figure 6

<hr />

